File: 2018/python/01.py
with open('../input/01.txt') as fp:
    x = sum([int(l) for l in fp])
    print('Part 1: {}'.format(x))

with open('../input/01.txt') as fp:
    instructions = [int(l) for l in fp]

frequency = 0
i = 0
seen = [frequency]
while True:
    frequency += instructions[i]

    if frequency in seen:
        print('Part 2: {}'.format(frequency))
        break

    seen.append(frequency)
    i = (i + 1) % len(instructions)

File: 2015/csharp/01.cs
class Day01 {
	static void Main() {
		string instructions = System.IO.File.ReadAllText(@"../input/input1.txt");
		int floor = 0;
		int pos = 1;
		int basement = -1;
		foreach (char dir in instructions) {
			if (dir == '(') {
				floor++;
			} else if (dir == ')') {
				floor--;
			} else {
				throw new System.Exception("how did this even happen");
			}
			if (basement == -1 && floor < 0) {
				basement = pos;
			} else {
				pos++;
			}
		}
		System.Console.WriteLine("Part 1: {0}", floor);
		System.Console.WriteLine("Part 2: {0}", basement);
	}
}
